src/uct/cuda/base/cuda_base.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCT_CUDA_BASE_H_
+#define UCT_CUDA_BASE_H_
+
+#include <uct/base/uct_iface.h>
+
+
+/**
+ * Query the list of available cuda devices.
+ *
+ * @param [in]  md               Memory domain to run the query on.
+ * @param [out] tl_devices_p     List of available devices on the md.
+ * @param [out] num_tl_devices_p Number of available devices on the md.
+ *
+ * @return UCS_OK if successful, or UCS_ERR_NO_MEMORY if failed to allocate the
+ *         array of device resources.
+ */
+ucs_status_t
+uct_cuda_base_query_devices(uct_md_h md,
```

Comment:
maybe move these funcs to cuda_md.h, instead of creating cuda_base.h?
it seems gdrcopy is already including cuda_md.h